index.html

```diff
@@ -5007,13 +4989,281 @@ <h3 id="looping-AudioBufferSourceNode">
+            of this operation is up to the UA. Examples of such operations
+            might include upsampling, downsampling, applying a subsample
+            offset, or loop unrolling. This operation may affect the values of
+            <code>sampleRate</code>, <code>playbackRate</code> and
+            <code>offset</code>.
+            </li>
+            <li>Let <code>bufferTime</code> equal -1. This is the <em>playhead
+            position</em> within <code>buffer</code> of the next output sample
+            frame. The initial value -1 indicates that this position has not
+            yet been determined.
+            </li>
+            <li>
+              <em>Optimize the buffer</em> prior to rendering, if desired. For
+              example, if <code>playbackRate</code> is a constant value other
+              than 1, resampling up front for this rate may eliminate the need
+              for interpolation later.
+            </li>
+            <li>For each block of audio to be rendered:
+              <ol>
+                <li>Let <code>currentTime</code> be the current time of the <a>
+                  AudioContext</a>.
+                </li>
+                <li>Let <code>dt</code> be 1 / (_context sample rate_).
+                </li>
+                <li>Let <code>index</code> be 0.
+                </li>
+                <li>Let <code>computedPlaybackRate</code> be <code>playbackRate
+                * pow(2, detune / 1200) * (bufferRate / contextRate)</code> .
+                This combines the two k-rate parameters affecting playback rate
+                into a single computed quantity, and also corrects for any
+                difference in sample rate between the buffer and the
+                <a>AudioContext</a>.
+                </li>
+                <li>
+                  <em>Optimize the buffer</em> prior to rendering the block, if
+                  desired.
+                </li>
+                <li>While <code>index</code> is less than the length of the
+                audio block to be rendered:
+                  <ol>
+                    <li>If <code>currentTime &lt; start</code> or
+                    <code>currentTime &gt;= stop</code>, emit silence for the
+                    output frame at <code>index</code>.
+                    </li>
+                    <li>Else,
+                      <ol>
+                        <li>If <code>bufferTime</code> &lt; 0, set
+                        <code>bufferTime</code> to <code>offset + ((currentTime
+                        - start) * computedPlaybackRate)</code>. This is the
+                        playhead position for the first output sample frame.
+                        </li>
+                        <li>Let <code>effectiveBufferTime</code> equal
+                        <code>bufferTime</code>.
+                        </li>
+                        <li>While <code>loop</code> is true and
```

Comment:
I find this reads a bit oddly.  Maybe have bullet 3 just be the text "map the playhead...", then a new list consisting of

1. while loop is true and effectiveBufferTime >= actualLoopEnd
    1. effectiveBufferTime -= actualLoopEnd - actualLoopStart